src/theme.c

```diff
@@ -452,6 +465,56 @@ parse_hexstrs(const char *hexes, float colors[3][4])
 	g_strfreev(elements);
 }
 
+static void
+parse_color(const char *str, float *rgba)
+{
+	if (str[0] == '#') {
+		parse_hexstr(str, rgba);
+	} else if (!strncasecmp(str, "rgb:", 4)) {
+		size_t len = strlen(str);
+		if (len == 9 && str[5] == '/' && str[7] == '/') {
+			/* rgb:R/G/B */
+			rgba[0] = (hex_to_dec(str[4]) * 16) / 255.0;
+			rgba[1] = (hex_to_dec(str[6]) * 16) / 255.0;
+			rgba[2] = (hex_to_dec(str[8]) * 16) / 255.0;
+			rgba[3] = 1.0;
+		} else if (len == 12 && str[6] == '/' && str[9] == '/') {
+			/* rgb:RR/GG/BB */
+			rgba[0] = (hex_to_dec(str[4]) * 16 + hex_to_dec(str[5])) / 255.0;
+			rgba[1] = (hex_to_dec(str[7]) * 16 + hex_to_dec(str[8])) / 255.0;
+			rgba[2] = (hex_to_dec(str[10]) * 16 + hex_to_dec(str[11])) / 255.0;
+			rgba[3] = 1.0;
+		}
+	} else {
+		uint32_t argb = 0;
+		if (lookup_named_color(str, &argb)) {
+			rgba[0] = ((argb >> 16) & 0xFF) / 255.0;
+			rgba[1] = ((argb >> 8) & 0xFF) / 255.0;
+			rgba[2] = (argb & 0xFF) / 255.0;
+			rgba[3] = 1.0;
+		}
+	}
+}
+
+static enum lab_gradient
+parse_gradient(const char *str)
```

Comment:
What did you base the parsing methodology on?
1. Openbox source code
2. An interpretation of the openbox wiki documentation: https://openbox.org/help/Themes#Textures
3. Reverse engineering
4. Other

Reply:
I had to look at the Openbox source code to see what it was doing, because the themes shipped with Openbox itself don't follow what one would expect from the wiki. Our `parse_gradient` is not a copy of the Openbox equivalent though (it's much simpler).

Reply:
:+1: 

Yes, I remember getting confused trying to understand the texture logic before. Think I was looking at the border part that time.